libraries/plugins/es_objects/es_objects.cpp

```diff
@@ -331,25 +352,25 @@ void es_objects_plugin_impl::PrepareBitAsset(const asset_bitasset_data_object& b
       bitasset.current_feed = fc::json::to_string(bitasset_object.current_feed);
       bitasset.current_feed_publication_time = bitasset_object.current_feed_publication_time;
 
-      auto it = bitassets.find(bitasset_object.id);
-      if(it == bitassets.end())
-         bitassets[bitasset_object.id] = bitasset;
-      else {
-         if(it->second == bitasset) return;
-         else bitassets[bitasset_object.id] = bitasset;
-      }
-
       std::string data = fc::json::to_string(bitasset);
 
       fc::mutable_variant_object bulk_header;
       bulk_header["_index"] = _es_objects_index_prefix + "bitasset";
       bulk_header["_type"] = "data";
+      if(_es_objects_keep_only_current)
+      {
+         bulk_header["_id"] = createIdstring(bitasset.object_id);
+      }
 
       prepare = graphene::utilities::createBulk(bulk_header, std::move(data));
-      bulk.insert(bulk.end(), prepare.begin(), prepare.end());
+      std::move(prepare.begin(), prepare.end(), std::back_inserter(bulk));
       prepare.clear();
    }
 }
+std::string es_objects_plugin_impl::createIdstring(object_id_type& object_id)
+{
+   return fc::to_string(object_id.space()) + "." + fc::to_string(object_id.type()) + "." + fc::to_string(object_id.instance());
```

Comment:
Why not use https://github.com/bitshares/bitshares-core/blob/2.0.180823/libraries/db/include/graphene/db/object_id.hpp#L86 instead? Keep your code DRY.